src/com/pb/javatraining/service/StatementService.java

```diff
@@ -15,66 +16,101 @@
 public class StatementService {
 
     private final Collection<Branch> data;
+    private List<Group> groups = new ArrayList<>();
+
 
     public StatementService() {
         DummyDataService dataService = new DummyDataService();
         this.data = dataService.getDummyData();
+        getGroups();
     }
 
-    public int getAmountOfGroupsWhereBadStudents() {
+    private void getGroups(){
+        groups = data.stream()
+                .flatMap(branch -> branch.getGroups().stream())
+                .collect(Collectors.toList());
+    }
 
-        // TODO
+    private List<Student> getAllStudents(){
+        return groups.stream()
+                .flatMap(group -> group.getStudents().stream())
+                .collect(Collectors.toList());
+    }
 
-        return 0;
+    public int getAmountOfGroupsWhereBadStudents() {
+        return (int) groups.stream()
+                .filter(group -> group.getStudents().stream()
+                        .mapToDouble(student -> student.getMarksByLessons().entrySet().stream()
+                                .mapToDouble(Map.Entry::getValue)
+                                .average()
+                                .getAsDouble())
+                        .anyMatch(average -> average < 60))
+                .count();
     }
 
-    public Map<Group, Integer> getAverageMarksWithinGroups() {
 
-        // TODO
 
-        return emptyMap();
+    public Map<Group, Double> getAverageMarksWithinGroups() {
+        return groups.stream().collect(Collectors.toMap(group -> group,
+                group ->  group.getStudents().stream()
+                        .mapToDouble(student -> student.getMarksByLessons().entrySet().stream()
+                                .mapToDouble(Map.Entry::getValue)
+                                .average()
+                                .getAsDouble())
+                        .average()
+                        .getAsDouble()));
     }
 
     public Collection<String> getGroupTitlesWhereStudentsAreMen() {
-
-        // TODO
-
-        return emptyList();
+       return groups.stream()
+                .filter(group -> group.getStudents().stream().
+                        allMatch(student -> student.getState() == State.MALE))
+                .map(Group::getTitle)
+                .collect(Collectors.toList());
     }
 
     public Collection<String> getStudentFullNamesFromSpecifiedBranch(Branch branch) {
-
-        // TODO
-
-        return emptyList();
+        return data.stream()
+                .filter(b -> b.equals(branch))
+                .flatMap(b -> b.getGroups().stream())
```

Comment:
try to always simplify the stream usage you develop, make it more atomic there.

`...data.stream().filter(Objects::equals).map(Branch::getGroups).flatMap(Stream::of)...`